### ISSUE TITLE
Site slowing down with MYSQL 8

### ISSUE THREAD

**Issue by robertgarrigos** [backdrop/backdrop-issues#4882]
I've seen this comment (https://github.com/backdrop/backdrop-issues/issues/4238#issuecomment-708487680) about mysql 8 being not slowing down backdrop, but I have to say that all my backdrop sites slowed down by too much just when upgraded my ubuntu server (20.04), which ,I believe, only works with mysql 8. Other sites I manage on ubuntu 18.04 work just as expected. 

Is someone else experiencing the same? I tuned up mysql 8 with no much luck.

**Comment by earlyburg** [comment]
Hi,
Can you tell us a bit more about your installation? Also, I am wondering if you are using any custom modules on the site in question?
Ubuntu 20.04 ships with MySql 8 and from https://github.com/backdrop/backdrop-issues/issues/4238 we can see _some_ differences in the way MySql 8 performs beside MySql 5.x with more complex queries but not so with brute force data access.
For example:

**bckdrp_mysql5:**
load front page
Page execution time was 115.35 ms.

load /admin/modules/list (modules listing page)
Page execution time was 134.41 ms.

load /admin/config/development/devel (devel setting page)
Page execution time was 54.38 ms.

**bckdrp_mysql8:**
load front page
Page execution time was 31.58 ms.

load /admin/modules/list (modules listing page)
Page execution time was 139.54 ms.

load /admin/config/development/devel (devel setting page)
Page execution time was 18.56 ms.

In these tests MySql8 performed much better than MySql5.

On these tests it was different:

**bckdrp_mysql5:**
load /admin/appearance/list (themes page)
Page execution time was 82.11 ms.

load /admin/config/search/settings (search settings page)
Page execution time was 61.49 ms.

load /admin/reports/fields/list (field list page)
Page execution time was 107.12 ms.

**bckdrp_mysql8:**
load /admin/appearance/list (themes page)
Page execution time was 126.89 ms.

load /admin/config/search/settings (search settings page)
Page execution time was 17.69 ms.

load /admin/reports/fields/list (field list page)
Page execution time was 156.49 ms.

It really just comes down to what type of site you are running and how your system is configured.
Further analysis may be necessary.

**Comment by domaingood** [comment]
I m using CentOS 8 Its already running MYSQL 8 working fine.

**Comment by earlyburg** [comment]
This method may be helpful for general analysis of speed issues.
Instal the Devel module
https://github.com/backdrop-contrib/devel/releases/tag/1.x-1.8.0
Navigate here:
/admin/config/development/devel
Check the checkbox that says "Display Page Timer", and save the setting.
You should be able to access and compare page load times on your installations.
I would love to hear feedback on this; the more data and feedback the better.

**Comment by robertgarrigos** [comment]
Yes, I'm using many contrib modules in some of my sites, but I have other basic sites with similar performance. The thing is that I had my backdrop sites with an ubuntu 18.04 server working perfectly at all. Then, I just upgraded it to ubuntu 20.04 and, suddenly, the same sites are just much slower. These are some examples:

Loading admin page:  ```/admin/structure/types/manage/page/fields```
```
Executed 107 queries in 2941.96 ms. Queries exceeding 5 ms are highlighted. Page execution time was 3189.85 ms. Memory used at: devel_boot()=0.58 MB, devel_shutdown()=5.72 MB, PHP peak=4 MB.
```

Saving the form in the same page:
```
Executed 705 queries in 19940.72 ms. Queries exceeding 5 ms are highlighted. Page execution time was 20315.04 ms. Memory used at: devel_boot()=0.56 MB, devel_shutdown()=21.12 MB, PHP peak=22 MB.
```
Some of the slower queries were:

```
1745.41 - 5 - BackdropDatabaseCache::set (link is external) - P A - 
INSERT INTO `cache_field` (`cid`, `serialized`, `created`, `expire`, `data`) VALUES (:db_insert_placeholder_0, :db_insert_placeholder_1, :db_insert_placeholder_2, :db_insert_placeholder_3, :db_insert_placeholder_4)  
4639.7 - 56 - BackdropDatabaseCache::set (link is external) - P A - 
INSERT INTO `cache_views` (`cid`, `serialized`, `created`, `expire`, `data`) VALUES (:db_insert_placeholder_0, :db_insert_placeholder_1, :db_insert_placeholder_2, :db_insert_placeholder_3, :db_insert_placeholder_4)  
1238.58 - 56 - BackdropDatabaseCache::set (link is external) - P A - 
INSERT INTO `cache_views` (`cid`, `serialized`, `created`, `expire`, `data`) VALUES (:db_insert_placeholder_0, :db_insert_placeholder_1, :db_insert_placeholder_2, :db_insert_placeholder_3, :db_insert_placeholder_4)
```
And dozens of other queries taking more than 100ms dealing with inserting into ```cache_views``` table.

I'll see if I have a backup of an old mysql 5 database data set and I can try to run the site with an ubuntu 18.04 for a while and compare results.

**Comment by earlyburg** [comment]
@robertgarrigos I am mindful that the Ubuntu version upgrade process has never been free of issues. I suggest that if you upgraded Ubuntu using the Ubuntu update script (sudo apt dist-upgrade) that if possible to backup your server and do a fresh install of Ubuntu 20.04.
https://unix.stackexchange.com/questions/128392/apt-get-script-to-update-upgrade-dist-upgrade-clean-and-autoremove

**Comment by robertgarrigos** [comment]
Thanks but I won't do a fresh install of the whole server....again. I rather find where the problem is. :-(

**Comment by earlyburg** [comment]
Agree. All we need to do is establish a baseline to compare to. I'm mindful that since your code did not change, and your server environment did, that that may point to an issue with the server environment. I look forward to learning more about this issue, and thanks for taking the time to explain it to me. Things like this can be really frustrating.

**Comment by robertgarrigos** [comment]
I just moved the backdrop install to a fresh ubuntu 20.04 server droplet and the figures are much better. What it took 19 seconds, now it takes 1.3 seconds on a much worse server, in terms of memory and cpu, but both with ubuntu 20.04. i thought the problem was the CPU, but it gets higher on the new droplet, which is normal, as it is a shared cpu. Even though, it's much faster.

Could this be a writing to disk problem? How could I check it out?

**Comment by indigoxela** [comment]
> I just moved the backdrop site to a fresh ubuntu 20.04 server droplet and the figures are much better.

@robertgarrigos you probably already did that, but comparing _all_ mysql variables between these two MySQL 8 installs and figure out, which ones differ, might help. (For instance, binary logging is a known performance killer.)

**Comment by robertgarrigos** [comment]
Thanks @indigoxela I didn't about the binary logging. I'll check it out.

**Comment by robertgarrigos** [comment]
Checked: log_bin is enabled by default in mysql8. It is enabled in both server. Disabling it in the slow one doesn't fix it, though. Thanks all for your help.
